.github/workflows/ruby_ci.yml

```diff
@@ -1,84 +1,22 @@
+name: CI
+
+on: pull_request
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
 jobs:
   rubocop:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - run: sudo apt-get update
-      - run: sudo apt-get install libyaml-dev
-      - name: Setup Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: 2.7
-          bundler-cache: true
-          cache-version: 1
-          rubygems: 3.0.0
-      - name: Run rubocop
-        if: github.event_name != 'push'
-        run: bundle exec rubocop --format github
-  test_ruby:
-    runs-on: ubuntu-latest
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: Ruby
     needs: rubocop
-    timeout-minutes: 30
-    env:
-      BUNDLE_WITHOUT: journald:development:console:libvirt
-    services:
-      postgres:
-        image: postgres:12.1
-        ports: ['5432:5432']
-        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
-    strategy:
-      fail-fast: false
-      matrix:
-        foreman-core-branch: [develop]
-        ruby-version: [2.7]
-        node-version: [12]
-    steps:
-      - run: sudo apt-get update
-      - run: sudo apt-get install build-essential libcurl4-openssl-dev zlib1g-dev libpq-dev libyaml-dev
-      - uses: actions/checkout@v2
-        with:
-          repository: theforeman/foreman
-          ref: ${{ matrix.foreman-core-branch }}
-      - uses: actions/checkout@v2
-        with:
-          path: foreman_remote_execution
-      - name: Setup Bundler
-        run: |
-          echo "gem 'foreman_remote_execution', path: './foreman_remote_execution'" > bundler.d/foreman_remote_execution.local.rb
-      - name: Setup Ruby
-        uses: ruby/setup-ruby@v1
-        with:
-          ruby-version: ${{ matrix.ruby-version }}
-          bundler-cache: true
-      - name: Setup Node
-        if: github.event_name != 'push'
-        uses: actions/setup-node@v1
-        with:
-          node-version:  ${{ matrix.node-version }}
-      - name: Prepare test DB
-        if: github.event_name != 'push'
-        run: |
-          bundle exec rake db:create
-          bundle exec rake db:migrate
-      - name: Run plugin tests
-        if: github.event_name != 'push'
-        run: |
-          bundle exec rake test:foreman_remote_execution
-          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
-      - name: 'Upload logs'
-        uses: actions/upload-artifact@v2
-        if: failure()
-        with:
-          name: logs
-          path: log/*.log
-          retention-days: 5
+    uses: theforeman/actions/.github/workflows/foreman_plugin.yml@v0
+    with:
+      plugin: foreman_remote_execution
+      # https://github.com/theforeman/foreman/pull/9871
+      foreman_repository: ofedoren/foreman
+      foreman_version: feat-36849-add-ruby-3-ga
```

Comment:
It should probably now test against https://github.com/theforeman/foreman/pull/9989